0007-python3.md

```diff
@@ -0,0 +1,97 @@
+* [Ubuntu 18.04 through 2023](https://wiki.ubuntu.com/Releases)
+* [RHEL/CentOS 7 through 2024](https://wiki.centos.org/About/Product)
+
+Archivematica is not fully compatible with Python 3 yet. Archivematica and all
+of its required components will eventually need to run in a Python 3
+environment, especially as downstream dependencies begin to drop support for
+Python 2 (i.e.
+[Django](https://docs.djangoproject.com/en/2.2/faq/install/#what-python-version-should-i-use-with-django)).
+
+Components:
+
+* [am/compose](https://github.com/artefactual-labs/am)
+* [archivematica-storage-service](https://github.com/artefactual/archivematica-storage-service)
+* [MCPServer](https://github.com/artefactual/archivematica/tree/a889605d8c97e114f8c0bc707d6a371030fb5c0b/src/MCPServer)
+* [MCPClient](https://github.com/artefactual/archivematica/tree/a889605d8c97e114f8c0bc707d6a371030fb5c0b/src/MCPClient)
+* [archivematica-common](https://github.com/artefactual/archivematica/tree/a889605d8c97e114f8c0bc707d6a371030fb5c0b/src/archivematicaCommon)
+* [Dashboard](https://github.com/artefactual/archivematica/tree/a889605d8c97e114f8c0bc707d6a371030fb5c0b/src/dashboard)
+* [mets-reader-writer](https://github.com/artefactual-labs/mets-reader-writer)
+* [ammcpc](https://github.com/artefactual-labs/ammcpc)
+* [agentarchives](https://github.com/artefactual-labs/agentarchives)
+* [fixity](https://github.com/artefactual/fixity)
+* [automation-tools](https://github.com/artefactual/automation-tools)
+* [archivematica-acceptance-tests](https://github.com/artefactual-labs/archivematica-acceptance-tests)
+* [amclient](https://github.com/artefactual-labs/amclient)
+
+## Decision drivers <!-- optional -->
+
+* Archivematica is built on Python 2
+* Python 2 will no longer be supported after January 1, 2020
+* Archivematica should be refactored to use Python 3.5 (at minimum)
+
+## Considered options
+
+1. Refactor to Python 3.5
+
+## Decision outcome
+
+Chosen option: 1, because refactoring Archivematica and all required components
+to Python 3 is the only option to ensure that Archivematica continues to be a
+safe, stable environment.
+
+Python 3.5 was chosen for the following reasons:
+
+* [Python 3.4](https://www.python.org/dev/peps/pep-0429/) has already been
+  retired
+* [Python 3.5](https://www.python.org/dev/peps/pep-0478/) is the oldest version
+  supported
+* Ubuntu 16.04 ships Python 3.5
+* Ubuntu 18.04 ships Python 3.6
+* For CentOS 7, The Software Collections Repository provides packages for
+  Python 3.5 and up
+
+[Epic for Python 3 refactoring](https://github.com/archivematica/Issues/issues/805)
+
+### Positive consequences
+
+* Archivematica will be upgraded to a maintained version of Python.
+* Archivematica developers will be able to take advantage of new features in
+  Python.
+* The most recent version of Python 3 at time of writing is Python 3.9, which
+  will be maintained until June 2025; further versions of Python 3 are expected
```

Comment:
It isn't clear to me why we're mentioning this here. Is 3.9 just the last version likely to be created? 

This is also quite far in the future. With 3.8 and 3.9 still in testing, we could say 3.7 is the latest stable feature release of Python and will be supported until 2023?

* 3.7: https://www.python.org/dev/peps/pep-0537/
* 3.8: https://www.python.org/dev/peps/pep-0569/
* 3.9: https://www.python.org/dev/peps/pep-0596/